### DISCUSSION TITLE
Docker Self-Host Migrations

### DISCUSSION THREAD

**Discussion by Samsonov0** [supabase/supabase#13152] [Questions]
My team and I use the supabase self-host solution, and we needed to move the database from one server to another, but as it turned out, all this time migrations on our server were simply not created, and all the changes made were sent directly to the database. How can I create migrations and use them in the future?Important: We don't use Sapabese CLI, only docker self-host solution

**Comment by sweatybridge** [comment]
Hello, the CLI supports `--db-url` flag that lets you create a schema dump from self-hosted database. You can then restore the schema to another database with psql.

**Reply by Samsonov0** [reply]
Thanks for the answer, but again, we DON'T use the supabase CLI

**Comment by Samsonov0** [comment]
We use a self-host docker from the supabase repository, where there is no supabase CLI to perform the actions from the comment above. I am interested in a solution related specifically to docker-compose without the participation of the CLI

**Reply by spinus** [reply]
I know it's bit old but still. You got the answer. You can use any other migration tool, supabase cli is just one of them. You don't supabase-cli? Why, what's wrong with it? If not, use any other migration tool of your choice. I guess nobody is responding to you as this is not part of the stack, if you decided not to use cli - it's your stuff to figure out migration path and tooling.

**Comment by Samsonov0** [comment]
???

**Comment by solomonBoltin** [comment]
Same issue here, I am trying to achive fully declaritev api project where i provide supabse config, a /schemas directory - and get full authenticated api 
the thing is its not clear how supabase cli integrates with the running docker-compose instance
I was trying to create a docker file that initilies supabse cli and runs commands to add add the schemas but that didnt work, and cli is only for development any way so i still need to find a valid way to  automate initial schema and migrations loading when selfhosting docker-compose starts

**Reply by Rubiniuz** [reply]
Did you figure out a way to do this. I started a discussion about using the supabase cli with a self-hosted docker setup:
https://github.com/orgs/supabase/discussions/35616

I really think being able to use the cli on you dev docker env could increase productivity.
but just like you stated. when everything is ready for prod you want to be able to run all those migrations. running them by hand could become quite a pain.
